freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,241 @@
+                print(
+                    f"Only found {str(found_signals)} buy signals in the full timerange from "
+                    f"{str(self.full_varHolder.from_dt)} to "
+                    f"{str(self.full_varHolder.to_dt)}. "
+                    f"will increase timerange trying to get at least "
+                    f"{str(self.signals_to_check)} signals.")
+            else:
+                print(
+                    f"Found {str(found_signals)} buy signals, more than necessary. "
+                    f"Reducing to {str(self.signals_to_check)} "
+                    f"checked buy signals in the full timerange from "
+                    f"{str(self.full_varHolder.from_dt)} to {str(self.full_varHolder.to_dt)}")
+                break
+        if not continue_with_strategy:
+            return
+
+        for idx, result_row in self.full_varHolder.result['results'].iterrows():
+            if self.current_analysis.total_signals == self.signals_to_check:
+                break
+
+            # if force-sold, ignore this signal since here it will unconditionally exit.
+            if result_row.close_date == self.dt_to_timestamp(self.full_varHolder.to_dt):
+                continue
+
+            self.current_analysis.total_signals += 1
+
+            self.entry_varholder = backtest_lookahead_bias_checker.varHolder()
+            self.exit_varholder = backtest_lookahead_bias_checker.varHolder()
+
+            self.entry_varholder.from_dt = self.full_varHolder.from_dt  # result_row['open_date']
+            self.entry_varholder.compared_dt = result_row['open_date']
+
+            # to_dt needs +1 candle since it won't buy on the last candle
+            self.entry_varholder.to_dt = result_row['open_date'] + \
+                timedelta(minutes=timeframe_to_minutes(self.config['timeframe']) * 2)
+
+            self.prepare_data(self.entry_varholder, [result_row['pair']])
+
+            # ---
+            # print("analyzing the sell signal")
+            # to_dt needs +1 candle since it will always sell all trades on the last candle
+            self.exit_varholder.from_dt = self.full_varHolder.from_dt  # result_row['open_date']
+            self.exit_varholder.to_dt = \
+                result_row['close_date'] + \
+                timedelta(minutes=timeframe_to_minutes(self.config['timeframe']))
+            self.exit_varholder.compared_dt = result_row['close_date']
+
+            self.prepare_data(self.exit_varholder, [result_row['pair']])
+
+            # register if buy signal is broken
+            if not self.report_signal(
+                    self.entry_varholder.result,
+                    "open_date", self.entry_varholder.compared_dt):
+                self.current_analysis.false_entry_signals += 1
+
+            # register if buy or sell signal is broken
+            if not self.report_signal(self.entry_varholder.result,
+                                      "open_date", self.entry_varholder.compared_dt) \
```

Comment:
can we ... not use `\` to separate if statements ... wrapping the whole condition in "()" will make it look way better, you'll see (not just here - but also below).